### ISSUE TITLE
accept strings in Automerge.Text.insertAt

### ISSUE THREAD

**Issue by josharian** [automerge/automerge#326]
I sometimes have to do bulk inserts into an Automerge.Text, e.g. when importing a new document.

The current API accepts inserts one character at a time: `text.insertAt(0, 'h', 'e', 'l', 'l', 'o')`. Doing this with bulk inserts I've hit the javascript limit on the number of arguments allowed to a function call (65536).

I'd like it if insertAt also accepted strings and treated them indistinguishably from individual characters: `text.insertAt(0, 'hello')` (and maybe `text.insertAt(0, 'he', 'llo')`, not sure).

Or alternatively, a new API that accepted only one single string: `text.insertStringAt(0, 'hello')`.

I'm willing to attempt an implementation if you're open to it.

**Comment by ept** [comment]
@josharian Sorry for the delay. I think this is a good idea, and we can make an API change for this in the 1.0 release.

Suggestion: how about we get rid of `insertAt`, and replace it with `insertCharsAt(index, chars)` and `insertStringAt(index, string)`? `chars` can be an array of characters (not varargs/spread syntax), and `string` is split into an array of characters using `[...string]` (see also #334).

The rationale for providing a character-wise insertion method is that if an application wants to use a different way of splitting a string into characters (e.g. by Unicode grapheme cluster rather than by code point), it can do so.

**Comment by ept** [comment]
If you do a PR for this, please make it relative to the performance branch, so that it goes in the 1.x release series.

**Comment by pvh** [comment]
Is there a good reason to make these separate methods rather than doing something pseudo-polymorphic and checking the input for array / string?

**Comment by ept** [comment]
@pvh Good point, that would be better. So maybe we have a single `insertAt` method as now, but have it accept an array of characters or a string (which gets split) as second argument?

**Comment by lightsprint09** [comment]
@pvh @josharian @ept 

I implemented a pseudo-polymorphic insertAt in #350.
 Please give some feedback and we can iterate on it.
